src/test/java/dog/pawbook/model/managedentity/owner/UniqueEntityListTest.java

```diff
@@ -70,20 +71,20 @@ public void setOwner_targetOwnerNotInList_throwsOwnerNotFoundException() {
         assertThrows(EntityNotFoundException.class, () -> uniqueEntityList.setEntity(1, ALICE));
     }
 
-    @Test
-    public void remove_nullOwner_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> uniqueEntityList.remove(null));
-    }
-
     @Test
     public void remove_ownerDoesNotExist_throwsOwnerNotFoundException() {
-        assertThrows(EntityNotFoundException.class, () -> uniqueEntityList.remove(ALICE));
+        assertThrows(EntityNotFoundException.class, () -> uniqueEntityList.remove(-1));
     }
 
     @Test
     public void remove_existingOwner_removesOwner() {
         uniqueEntityList.add(ALICE);
-        uniqueEntityList.remove(ALICE);
+
+        List<Pair<Integer, Entity>> targets = uniqueEntityList.asObservableList().stream()
```

Comment:
okay I think i get the rationale for the returning the internalList directly, but it is alright to use the unmodifiable version as many other tests does the same. Also I think if this is targeted at owners but will not test dogs, it is a good idea to filter using instanceof as well.

Reply:
Edited